src/EmitterInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Runner\Http;
+
+use Psr\Http\Message\ResponseInterface;
+use Yiisoft\Yii\Runner\Http\Exception\HeadersHaveBeenSentException;
+
+/**
+ * `EmitterInterface` is responsible for sending HTTP responses.
+ */
+interface EmitterInterface
+{
+    /**
+     * Sends the response to the client with headers and body.
+     *
+     * @param ResponseInterface $response Response object to send.
+     *
+     * @throws HeadersHaveBeenSentException
+     */
+    public function emit(ResponseInterface $response): void;
```

Comment:
```suggestion
    public function emit(ResponseInterface $response, bool $withoutHeaders): void;
```

I think we need an ability to force not emitting the body for `HEAD` responses.